tests/install_upgrade_operators/network_policy/test_network_policy_components.py

```diff
@@ -0,0 +1,289 @@
+    This ensures that without restrictive policies, the connection should succeed.
+    """
+    with AllowAllNetworkPolicy(
+        name="test-allow-all-traffic",
+        namespace=hco_namespace.name,
+        client=admin_client,
+        match_labels=network_policy_match_labels,
+    ):
+        pod_name = pods_by_network_policy[0]
+        service_ip = deployed_server_service.instance.spec.get("clusterIP")
+        if ":" in service_ip:
+            pytest.fail(f"This test does not support sending traffic to IPv6: {service_ip}")
+
+        # Get Pod object to use execute() method
+        test_pod = Pod(client=admin_client, namespace=hco_namespace.name, name=pod_name)
+        try:
+            test_pod.execute(
+                command=shlex.split(f"curl -sS --connect-timeout 5 http://{service_ip}:{TEST_SERVER_PORT}")
+            )
+        except Exception as e:
+            pytest.fail(
+                f"Egress connectivity test preparation FAILED: With allow-all NetworkPolicy, "
+                f"connection from {pod_name} to test server pod should succeed but got error: {e}"
+            )
+
+    yield {
+        "pods_by_network_policy": pods_by_network_policy,
+        "hco_namespace": hco_namespace,
+        "server_service_ip": service_ip,
+    }
+
+
+@pytest.mark.polarion("CNV-12305")
+@pytest.mark.parametrize("network_policy_name", [KUBEVIRT_CONSOLE_PLUGIN_NP])
+def test_console_plugin_network_policy_blocks_ingress_unauthorized_traffic(ingress_connectivity_baseline):
+    """
+    Test that KUBEVIRT_CONSOLE_PLUGIN_NP blocks unauthorized traffic.
+
+    It uses a test pod with curl to try to reach the service IP+PORT,
+    which should be blocked by the NetworkPolicy.
+    """
+    deployed_client_pod = ingress_connectivity_baseline["deployed_client_pod"]
+    service_ip = ingress_connectivity_baseline["service_ip"]
+    service_port = ingress_connectivity_baseline["service_port"]
+
+    with pytest.raises(ExecOnPodError):
+        deployed_client_pod.execute(
+            command=shlex.split(f"curl -sS -k --connect-timeout 5 https://{service_ip}:{service_port}")
+        )
+
+
+@pytest.mark.polarion("CNV-12306")
+@pytest.mark.parametrize("network_policy_name", [KUBEVIRT_APISERVER_PROXY_NP])
+def test_apiserver_proxy_network_policy_blocks_egress_unauthorized_traffic(
+    admin_client,
+    egress_connectivity_baseline,
+):
+    """
+    Test that KUBEVIRT_APISERVER_PROXY_NP blocks unauthorized traffic.
+
+    It iterates through all pods matching the NetworkPolicy podSelector
+    and verifies that connections to the console plugin service are blocked.
+    """
+    pods_by_network_policy = egress_connectivity_baseline["pods_by_network_policy"]
+    hco_namespace = egress_connectivity_baseline["hco_namespace"]
+    server_service_ip = egress_connectivity_baseline["server_service_ip"]
+
+    for pod_name in pods_by_network_policy:
```

Comment:
any reason to iterate over all pods? they are managed by a deployment and will have the same behavior